src/main/java/org/jitsi/videobridge/cc/CcConfig.java

```diff
@@ -0,0 +1,174 @@
+        protected static final String propName = "videobridge.cc.onstage-preferred-height-px";
+
+        protected OnstagePreferredHeightProperty()
+        {
+            super(new JvbPropertyConfig<Integer>()
+                .fromLegacyConfig(config -> config.getInt(legacyPropName))
+                .fromNewConfig(config -> config.getInt(propName))
+                .readOnce()
+                .throwIfNotFound()
+            );
+        }
+    }
+
+    public static OnstagePreferredHeightProperty onstagePreferredHeight = new OnstagePreferredHeightProperty();
+
+    /**
+     * The property of the preferred frame rate to allocate for the onstage
+     * participant.
+     */
+    public static class OnstagePreferredFramerateProperty extends AbstractConfigProperty<Double>
+    {
+        protected static final String legacyPropName = "org.jitsi.videobridge.ONSTAGE_PREFERRED_FRAME_RATE";
+        protected static final String propName = "videobridge.cc.onstage-preferred-framerate-fps";
+
+        protected OnstagePreferredFramerateProperty()
+        {
+            super(new JvbPropertyConfig<Double>()
+                .fromLegacyConfig(config -> config.getDouble(legacyPropName))
+                .fromNewConfig(config -> config.getDouble(propName))
+                .readOnce()
+                .throwIfNotFound()
+            );
+        }
+    }
+
+    public static OnstagePreferredFramerateProperty onstagePreferredFramerate = new OnstagePreferredFramerateProperty();
+
+    /**
+     * The property of the option that enables/disables video suspension
+     * for the on-stage participant.
+     */
+    public static class OnstageVideoSuspensionEnabledProperty extends AbstractConfigProperty<Boolean>
+    {
+        protected static final String legacyPropName = "org.jitsi.videobridge.ENABLE_ONSTAGE_VIDEO_SUSPEND";
+        protected static final String propName = "videobridge.cc.onstage-video-suspension-enabled";
+
+        protected OnstageVideoSuspensionEnabledProperty()
+        {
+            super(new JvbPropertyConfig<Boolean>()
+                .fromLegacyConfig(config -> config.getBoolean(legacyPropName))
+                .fromNewConfig(config -> config.getBoolean(propName))
+                .readOnce()
+                .throwIfNotFound()
+            );
+        }
+    }
+
+    public static OnstageVideoSuspensionEnabledProperty onstageVideoSuspensionEnabled = new OnstageVideoSuspensionEnabledProperty();
+
+    /**
+     * The property used to trust bandwidth estimations.
+     */
+    public static class TrustBweProperty extends AbstractConfigProperty<Boolean>
+    {
+        protected static final String legacyPropName = "org.jitsi.videobridge.TRUST_BWE";
+        protected static final String propName = "videobridge.cc.trust-bwe";
+
+        protected TrustBweProperty()
+        {
+            super(new JvbPropertyConfig<Boolean>()
```

Comment:
Can we introduce an abstraction for "a property with a legacy name that is read once"?

We could then have something like
```
public static AbstractConfigProperty<Boolean> trustBwe = LegacyNameProperty<Boolean>(
        "org.jitsi.videobridge.TRUST_BWE",
        "videobridge.cc.trust-bwe");
```

Same goes for the other 6 properties in this file (and I expect other files as well)